.github/workflows/build-wheels.yml

```diff
@@ -67,36 +67,192 @@ jobs:
 
       - name: build metatensor-core wheel
         run: python -m cibuildwheel python/metatensor-core
         env:
           CIBW_BUILD: cp311-*
           CIBW_SKIP: "*musllinux*"
-          CIBW_ARCHS: ${{ matrix.cibw_arch }}
+          CIBW_ARCHS: ${{ matrix.cibw-arch }}
           CIBW_BUILD_VERBOSITY: 2
           CIBW_MANYLINUX_X86_64_IMAGE: rustc-manylinux2014_x86_64
           CIBW_MANYLINUX_AARCH64_IMAGE: rustc-manylinux2014_aarch64
+          CIBW_ENVIRONMENT: >
+            METATENSOR_NO_LOCAL_DEPS=1
+            MACOSX_DEPLOYMENT_TARGET=11
 
       - name: check wheels with twine
         run: twine check wheelhouse/*
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
-          name: wheels
+          name: core-wheel-${{ matrix.os }}-${{ matrix.cibw-arch }}
           path: ./wheelhouse/*.whl
 
-      - name: upload wheel to GitHub release
-        if: startsWith(github.ref, 'refs/tags/metatensor-core-v')
-        uses: softprops/action-gh-release@v1
+  build-torch-wheels:
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.name }} (torch v${{ matrix.torch-version }})
+    strategy:
+      matrix:
+        torch-version: ['1.11', '1.12', '1.13', '2.0', '2.1', '2.2']
+        arch: ['arm64', 'x86_64']
+        os: ['ubuntu-22.04', 'macos-11', 'macos-14', 'windows-2019']
+        exclude:
+          # remove mismatched arch for macOS
+          - {os: macos-14, arch: x86_64}
+          - {os: macos-11, arch: arm64}
+          # no arm64-windows build
+          - {os: windows-2019, arch: arm64}
+          # arm64-macos is only supported for torch >= 2.0
+          - {os: macos-14, arch: arm64, torch-version: '1.11'}
+          - {os: macos-14, arch: arm64, torch-version: '1.12'}
+          - {os: macos-14, arch: arm64, torch-version: '1.13'}
+        include:
+          # add `cibw-arch` and `rust-target` to the different configurations
+          - name: x86_64 Linux
+            os: ubuntu-22.04
+            arch: x86_64
+            rust-target: x86_64-unknown-linux-gnu
+            cibw-arch: x86_64
+          - name: arm64 Linux
+            os: ubuntu-22.04
+            arch: arm64
+            rust-target: aarch64-unknown-linux-gnu
+            cibw-arch: aarch64
+          - name: x86_64 macOS
+            os: macos-11
+            arch: x86_64
+            rust-target: x86_64-apple-darwin
+            cibw-arch: x86_64
+          - name: arm64 macOS
+            os: macos-14
+            arch: arm64
+            rust-target: aarch64-apple-darwin
+            cibw-arch: arm64
+          - name: x86_64 Windows
+            os: windows-2019
+            arch: x86_64
+            rust-target: x86_64-pc-windows-msvc
+            cibw-arch: AMD64
+          # add the right python version for each torch version
+          - {torch-version: '1.11', python-version: '3.10', cibw-python: 'cp310-*'}
```

Comment:
Can't you run this with Py3.12? If yes does this work? I thought we have to build the wheels for every Python major version.

Reply:
no, there are no pre-built torch 1.11 for python 3.12. The versions here are the highest Python versions for which there is a prebuilt torch wheel

Reply:
Okay I understand.